File: src/main/java/com/qtech/bubbles/bubble/HealBubble.java
package com.qtech.bubbles.bubble;

import com.qtech.bubbles.common.ResourceLocation;
import com.qtech.bubbles.common.entity.Entity;
import com.qtech.bubbles.core.utils.categories.ColorUtils;
import com.qtech.bubbles.entity.BubbleEntity;
import com.qtech.bubbles.entity.player.PlayerEntity;
import org.apache.commons.lang.math.DoubleRange;
import org.apache.commons.lang.math.IntRange;

import java.awt.*;

public class HealBubble extends AbstractBubble {
//    public Color[] colors;

    public HealBubble() {
        colors = ColorUtils.multiConvertHexToRgb("#ffffff");
        colors = new Color[]{new Color(0, 192, 0), new Color(0, 0, 0, 0), new Color(0, 192, 0), new Color(0, 192, 0)};
        setRegistryName(ResourceLocation.fromString("bubbleblaster:heal_bubble"));

        setPriority(4000000);
        setRadius(new IntRange(17, 70));
        setSpeed(new DoubleRange(10.2d, 18.6d));
        setDefense(0.3f);
        setAttack(0.0f);
        setScore(1);
        setHardness(1.0d);

//        BubbleInit.BUBBLES.add(this);
    }

    @Override
    public void onCollision(BubbleEntity source, Entity target) {
        super.onCollision(source, target);

        if (target instanceof PlayerEntity) {
            PlayerEntity player = (PlayerEntity) target;
            player.restoreDamage(1.8f * (source.getGameType().getLocalDifficulty() / 20.0f + (1.8f / 20.0f)));
        }
    }
}

File: src/main/kotlin/qtech/bubbles/bubble/HealBubble.kt
package qtech.bubbles.bubble

import qtech.bubbles.common.ResourceLocation
import qtech.bubbles.entity.Entity
import qtech.bubbles.core.utils.categories.ColorUtils
import qtech.bubbles.entity.BubbleEntity
import qtech.bubbles.entity.player.PlayerEntity
import org.apache.commons.lang.math.DoubleRange
import org.apache.commons.lang.math.IntRange
import java.awt.Color

class HealBubble : AbstractBubble() {
    override fun onCollision(source: BubbleEntity, target: Entity) {
        super.onCollision(source, target)
        if (target is PlayerEntity) {
            target.restoreDamage(1.8f * (source.gameMode.localDifficulty / 20.0f + 1.8f / 20.0f))
        }
    }

    //    public Color[] colors;
    init {
        colors = ColorUtils.multiConvertHexToRgb("#ffffff")
        colors = arrayOf(Color(0, 192, 0), Color(0, 0, 0, 0), Color(0, 192, 0), Color(0, 192, 0))
        registryName = ResourceLocation.fromString("bubbleblaster:heal_bubble")
        priority = 4000000.0
        radius = IntRange(17, 70)
        speed = DoubleRange(10.2, 18.6)
        defense = 0.3f
        attack = 0.0f
        score = 1f
        hardness = 1.0

//        BubbleInit.BUBBLES.add(this);
    }
}